subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -63,10 +58,17 @@
      */
     Optional<? extends Iterable<String>> getChangingOutputs();
 
+    /**
+     * Whether files added are to be considered part of the output of the work.
```

Comment:
I think it is not about whether files added to the output locations are considered an output of the work. When the task adds new files in an output directory, they are always considered an output of the work.

It is about whether the task is considered up-to-date if something else adds outputs to the output locations, without changing the recorded outputs of the last execution of the work. So it is about whether to allow overlapping outputs or not, right?

Reply:
Clarified this a bit.